generators/fusesoc

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import yaml
+import subprocess
+
+templ = """/*
+:name: {0}
+:description: {1}
+:files: {2}
+:incdirs: {3}
+:top_module: {4}
+:tags: {5}
+:timeout: {6}
+*/
+"""
+
+try:
+    third_party_dir = os.environ['THIRD_PARTY_DIR']
+    tests_dir = os.environ['TESTS_DIR']
+    conf_dir = os.environ['CONF_DIR']
+except KeyError:
+    print("Export the THIRD_PARTY_DIR, TESTS_DIR and CONF_DIR variables first")
+    sys.exit(1)
+
+try:
+    tests_subdir = sys.argv[1]
+except IndexError:
+    print("Usage: ./generator <subdir>")
+    sys.exit(1)
+
+fusesoc_conf_files = os.listdir(
+    os.path.abspath(os.path.join(conf_dir, "fusesoc-configs")))
+
+for fusesoc_conf_file in fusesoc_conf_files:
+    fusesoc_conf_file = os.path.abspath(
+        os.path.join(conf_dir, "fusesoc-configs", fusesoc_conf_file))
+
+    with open(fusesoc_conf_file, 'r') as stream:
+        try:
+            yml_conf = yaml.safe_load(stream)
+        except yaml.YAMLError as err:
+            print(err)
+
+    sources = ''
+    incdirs = ''
+    incs = []
+    command_list = list(yml_conf["command"].split(" "))
```

Comment:
What if one of these keys is unavailable? Maybe we should catch it and report a meaningful message?